### PR TITLE
Escape negative numbers in integer-backed enums

### LINKED ISSUE #265: Integer backed enums can't handle negative numbers
Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift needs a fix, we should prefix negative numbers with e.g. "n", so "1" has the case "_1" and "-1" would have "_n1" or just "n1".

Right now if a negative number is used, the generated code won't compile.

### PR DESCRIPTION
**Pull request by PARAIPAN9**
### Motivation

Fixes #265

### Modifications

When handling a negative number, it takes its absolute value and add 'n' before it

### Result

Example: "-1" -> "n1"

### Test Plan

Ran and passed all the tests

### CODE REVIEW

File: Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift [hunk 1]
```diff
@@ -61,7 +61,7 @@ extension FileTranslator {
                     guard let rawValue = anyValue as? Int else {
                         throw GenericError(message: "Disallowed value for an integer enum '\(typeName)': \(anyValue)")
                     }
-                    let caseName = "_\(rawValue)"
+                    let caseName = rawValue < 0 ? "n\(abs(rawValue))" : "_\(rawValue)"
```
Comment (czechboy0) [inline Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift:64]: Thoughts on whether we should also make positive numbers spelled "p1" for consistency or keep the current "_1", @glbrntt @simonjbeaumont @gjcairo?

**Comment by czechboy0** [comment]
Please add a test case for this as well, just to be positive (ha) that it works.

File: Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift (hunk 1 above)
Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift:64]: That said, it'd be a breaking change, and I don't find `p1` significantly better than `_1`, so I'm also fine with keeping it as-is, and only fixing negative numbers, which today lead to non-compiling code, so no breaking change there.

Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift:64]: Ok, let's go with what @simonjbeaumont suggested. @PARAIPAN9 please update the PR with that and add a few unit tests and we should be set! 🙏

Comment (PARAIPAN9) [reply Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/translateStringEnum.swift:64]: Perfect, thank you for the clarifications!!

**Review by czechboy0 — APPROVED** [review]
Great, thanks!

**Pull request merged**